src/lambda_function/s3_event_config/app.py

```diff
@@ -56,35 +161,117 @@ def add_notification(
+    }
+
+    (
+        existing_bucket_notification_configuration,
+        existing_notification_configurations_for_type,
+    ) = get_existing_bucket_notification_configuration_and_type(
+        s3_client, bucket, destination_type
     )
-    logger.info("Put request completed....")
 
+    (
+        index_of_matching_arn,
+        matching_notification_configuration,
+    ) = get_matching_notification_configuration(
+        destination_type_arn,
+        existing_notification_configurations_for_type,
+        destination_arn,
+    )
 
-def delete_notification(s3_client: boto3.client, bucket: str):
-    """Deletes the S3 notification configuration from an existing bucket
+    if index_of_matching_arn is not None:
+        if json.dumps(
+            matching_notification_configuration, sort_keys=True
+        ) != json.dumps(new_notification_configuration, sort_keys=True):
+            existing_notification_configurations_for_type[
+                index_of_matching_arn
+            ] = new_notification_configuration
+            update_required = True
+    else:
+        existing_notification_configurations_for_type.append(
+            new_notification_configuration
+        )
+        update_required = True
+
+    if update_required:
+        existing_bucket_notification_configuration[
+            f"{destination_type}Configurations"
+        ] = existing_notification_configurations_for_type
+        s3_client.put_bucket_notification_configuration(
+            Bucket=bucket,
+            NotificationConfiguration=existing_bucket_notification_configuration,
+        )
+        logger.info(
+            f"Put request completed to add a NotificationConfiguration for"
+            + create_formatted_message(bucket, destination_type, destination_arn)
+        )
+    else:
+        logger.info(
+            f"Put not required as an existing NotificationConfiguration already exists for"
+            + create_formatted_message(bucket, destination_type, destination_arn)
+        )
+
+
+def delete_notification(
+    s3_client: boto3.client, bucket: str, destination_type: str, destination_arn: str
+) -> None:
+    """Deletes the S3 notification configuration from an existing bucket for a specific destination type.
 
     Args:
         s3_client (boto3.client) : s3 client to use for s3 event config
         bucket (str): bucket name of the s3 bucket to delete the config in
+        destination_type (str): String name of the destination type for the configuration
     """
-    s3_client.put_bucket_notification_configuration(
-        Bucket=bucket, NotificationConfiguration={}
+    destination_type_arn = f"{destination_type}Arn"
+
+    (
+        existing_bucket_notification_configuration,
+        existing_notification_configurations_for_type,
+    ) = get_existing_bucket_notification_configuration_and_type(
+        s3_client, bucket, destination_type
+    )
+
+    (
+        index_of_matching_arn,
+        matching_notification_confiugration,
```

Comment:
Not that it matters, but there's a typo here. What do you think of using `_` instead to indicate that we don't use this variable later in the code?